galsim/config/extra_psf.py

```diff
@@ -17,62 +17,100 @@
 #
 
 import galsim
+import math
+import numpy
+import logging
 
-def DrawPSFStamp(psf, config, bounds, offset, method):
+# The psf extra output type builds an Image of the PSF at the same locations as the galaxies.
+
+# The code the actually draws the PSF on a postage stamp.
+def DrawPSFStamp(psf, config, base, bounds, offset, method, logger=None):
     """
     Draw an image using the given psf profile.
 
     @returns the resulting image.
     """
-
-    if not psf:
-        raise AttributeError("DrawPSFStamp requires psf to be provided.")
-
-    if ('output' in config and 'psf' in config['output'] and 
-        'draw_method' in config['output']['psf'] ):
-        method = galsim.config.ParseValue(config['output']['psf'],'draw_method',config,str)[0]
+    if 'draw_method' in config:
+        method = galsim.config.ParseValue(config,'draw_method',base,str)[0]
         if method not in ['auto', 'fft', 'phot', 'real_space', 'no_pixel', 'sb']:
             raise AttributeError("Invalid draw_method: %s"%method)
     else:
         method = 'auto'
 
-    # Special: if the galaxy was shifted, then also shift the psf 
-    if 'shift' in config['gal']:
-        gal_shift = galsim.config.GetCurrentValue(config['gal'],'shift')
-        if False:
-            logger.debug('obj %d: psf shift (1): %s',config['obj_num'],str(gal_shift))
+    # Special: if the galaxy was shifted, then also shift the psf
```

Comment:
This was existing functionality that I preserved.  It's only relevant for demo5 I think, where we draw the psfs in the output image at the same (shifted) position as the galaxies.  However, I'd be happy to revisit this, since it's not necessarily what the user would probably expect.

Maybe have a special option for output.psf.shift = 'galaxy' or 'match' meaning to match the galaxy's shift, but otherwise allow any normal shift instead.  So demo5.yaml's output field would look like:
```
output :
    ...

    psf :
        file_name : g08_psf.fits
        shift : galaxy
```
Currently, `offset` works the same way.  If there is an `offset` parameter in config['image'] or config['stamp'], it applies here as well.  However, this doesn't actually show up in any of the demos currently.  We could make that work the same way with an `offset : galaxy` option, or allow a different offset value if the user specifies something else.

And if neither `offset` nor `shift` are in the `output.psf` field, then just leave the PSF centered.

That seems like a sensible behavior.  Do you agree?  It's a non-backward compatible API change, but I'm ok with that.